contents/blog/open-source-telemetry-ethical.md

```diff
@@ -0,0 +1,98 @@
+
+### Utilitarian-based ethics
+
+Moral behavior means opting for the greatest common good. An act is moral if it increases the happiness or decreases the misery of the greatest number of people. The results are the measure of moral behavior, not the intent.
+
+Stealing those chips? Go for it.
+
+## So, is it ok for open source projects to track users?
+
+Open source projects are zero expectation - they may contain bugs, security risks, or have confusingly outdated documentation. Remember that time you lost hours on a project that just didn't work in the first place?
+
+The fact it's acceptable for open source software not to work is a _great_ thing - it makes it easier for anyone to contribute. Mistakes, or half-baked projects included. You can't have innovation without a lot of people being able to contribute and make mistakes, or give up.
+
+From a utilitarian perspective, we believe that tracking the minimum data possible to build useful technology, in a way that doesn't share that data with 3rd parties, means that you have the right approach to product analytics in your open source project.
+
+From a duty-based perspective, we'd argue that transparency is key to covering intent. Since most people won't assume they're being tracked, that means being upfront in the docs and not relying on people digging through your code.
+
+## Minimizing your tracking
+
+### What you track
+
+The first step is to work out how much, if anything, you really need to track.
+
+#### Nothing
+
+Nada, zilch. You've no idea what's going on.
+
+If you're building it for fun, for yourself only, or to get better at a particular technology, then we don't think putting extra tracking in makes much sense.
+
+In fact, you'll struggle not to get some sense of popularity - issues or GitHub stars, and the basic stats (clones / repo visits) that GitHub provides if that's your preferred place to put code!
+
+#### Aggregated tracking
+
+This helps you validate if the concept of your project is valuable - do people install your thing? Do they actually use it at all?
+
+It also means you can get a sense of the setup flow. Can anyone work out how to implement things? At PostHog, early on, we tracked if people were able to get through the setup process. Many didn't, so we've kept tweaking it - and we still have work to do here :)
+
+Unfortunately, you cannot work out the unique users, which means repeat visitors are likely to skew your statistics and understanding. You could just have one person manically refreshing things!
+
+#### Anonymized user tracking
+
+This gives a much stronger signal than the above. It means you'll probably need to start using cookies.
+
+It allows you to see if people who started using the product are coming back, and which parts of the product they're returning too.
+
+#### Full user tracking
+
+Asking permission and not doing this by default feels like the only acceptable way.
```

Comment:
er

Reply:
@yakkomajuri to update our telemetry guidance